GPipe-GLFW4/test/Graphics/GPipe/BufferSpec.hs

```diff
@@ -20,3 +20,11 @@ spec = do
         writeBuffer buf 0 [0..50000]
         return $ bufferLength buf
       l `shouldBe` 100000
+
+  describe "resizeBuffer" $ do
+    it "should return the new size of the buffer" $ do
+      l <- runContextT GLFW.defaultHandleConfig $ do
+        buf :: Buffer os (B Float) <- newBuffer 0
+        resized <- resizeBuffer buf 100000
+        return $ bufferLength resized
+      l `shouldBe` 100000
```

Comment:
Can you add a few more tests here, like:
- try writing to the full length of the array
- try reading it back
- try writing more than the length of the array (catch exception, make sure it throws and doesn't just segfault)

Reply:
Now writes full length and also tests writing out of bounds (the one bellow). This is silently ignored which is a little weird.